### PR TITLE
Add agentless mode docs for Ruby library in CI - Test Visibility

### PR DESCRIPTION
**Pull request by anmarchenko**
### What does this PR do? What is the motivation?
We released agentless mode for Ruby CI visibility library: this PR adds documentation on how to setup agentless mode reusing the same docs as other libraries use. We remove legacy documentation on using build services to run datadog agent on cloud 

### Merge instructions
- [X] Merge after approval

### Additional notes
None

### CODE REVIEW

**Review by albertvaka — APPROVED** [review]

File: content/en/continuous_integration/tests/ruby.md [hunk 1]
```diff
@@ -502,11 +264,6 @@ All other [Datadog Tracer configuration][6] options can also be used.
 
 {{< partial name="whats-next/whats-next.html" >}}
 
-
-[1]: /agent/
-[2]: https://docs.datadoghq.com/agent/cluster_agent/admission_controller/
-[3]: https://app.datadoghq.com/organization-settings/api-keys
-[4]: /tracing/trace_collection/dd_libraries/ruby/#installation
```
Comment (hestonhoffman) [inline content/en/continuous_integration/tests/ruby.md:509]: Link 4 is still needed, it's referenced on line 72

**Review by hestonhoffman — COMMENTED** [review]
Looks good besides one broken link

File: content/en/continuous_integration/tests/ruby.md (hunk 1 above)
Comment (anmarchenko) [reply content/en/continuous_integration/tests/ruby.md:509]: @hestonhoffman fixed the link, thanks!

**Review by hestonhoffman — APPROVED** [review]

**Comment by hestonhoffman** [comment]
@anmarchenko is this one good to merge? I can fix the merge conflict and merge if it's ready

**Comment by anmarchenko** [comment]
@hestonhoffman yes, it is good to go!

**Pull request merged**